lib/utils/bit_list.ex

```diff
@@ -0,0 +1,92 @@
+defmodule LambdaEthereumConsensus.Utils.BitList do
+  @moduledoc """
+    Set of utilities to interact with bit list, represented as {bitstring, len}.
+  """
+  @type t :: {bitstring, integer()}
+  @bits_per_byte 8
+
+  @doc """
+  Creates a new bit_list from bitstring.
+  """
+  @spec new(bitstring) :: t
+  def new(bitstring) when is_bitstring(bitstring) do
+    # Change the byte order from little endian to big endian (reverse bytes).
+    num_bits = bit_size(bitstring)
+    len = length_of_bitlist(bitstring)
+
+    <<pre::integer-little-size(num_bits - 8), last_byte::integer-little-size(@bits_per_byte)>> =
+      bitstring
+
+    decoded = <<pre::integer-size(num_bits - 8), remove_trailing_bit(<<last_byte>>)::bitstring>>
```

Comment:
This doesn't look right, the last byte should end up as the first one after removing the trailing bit. To test for this, let's add some e2e (encode-decode) tests with a bitlist of size 10.